src/docfx/build/xref/XrefResolver.cs

```diff
@@ -47,7 +49,10 @@ internal class XrefResolver
         _redirectionProvider = redirectionProvider;
         _dependencyMapBuilder = dependencyMapBuilder;
         _fileLinkMapBuilder = fileLinkMapBuilder;
-        _xrefHostName = string.IsNullOrEmpty(config.XrefHostName) ? config.HostName : config.XrefHostName;
+        _hostNameMap = config.HostNameMapping ?? new Dictionary<string, string>();
+        _reverseHostNameMap = _hostNameMap.ToDictionary(x => x.Value, x => x.Key);
+        _xrefHostName = string.IsNullOrEmpty(config.XrefHostName) ? HostNameUtility.MigrateHostName(config.HostName, _hostNameMap)
```

Comment:
Does `_xrefHostName` control the output of XREF map? If so, we cannot use mapped host name here at this moment.